src/java.desktop/windows/classes/sun/awt/windows/WPrinterJob.java

```diff
@@ -1157,6 +1157,10 @@ protected boolean selectStylePen(int cap, int join, float width,
     protected boolean setFont(String family, float size, int style,
                               int rotation, float awScale) {
 
+        if (family.isEmpty()) {
```

Comment:
Not sure that the non-empty family, but spaces only will work.

Reply:
Would it be better to use isBlank() instead of isEmpty() to check a font family name is blank?
```
        if (family.isBlank()) {
            return false;
        }
```

Reply:
I think so.

Reply:
I changed the test SampleBow font family name to one space [SampleBowOneSpaceFamilyName.ttf](https://bugs.openjdk.java.net/secure/attachment/94402/SampleBowOneSpaceFamilyName.ttf) and to two spaces [SampleBowTwoSpacesFamilyName.ttf](https://bugs.openjdk.java.net/secure/attachment/94403/SampleBowTwoSpacesFamilyName.ttf)

The  PrintFontSample sample program properly prints text drawn by Graphics2D.drawString() and  Graphics2D.drawGlyphVector() on Windows with jdk 16 with these two fonts.
It looks like the problem is only with selecting a font in GDI with empty font family name.